### PR TITLE
Add selecteph to to pick the ephemeris of a specific bodies

### PR DESCRIPTION
**Pull request by LuEdRaMo**
I suggest to add a function `selecteph` to pick the ephemeris of a specific body. This can be helpful if we don't want to work with the full Solar System.

### CODE REVIEW

**Review by lbenet — COMMENTED** [review]
Thanks @LuEdRaMo for this addition! I just left a comment which should be very easy to address. Do you think it is possible to have a test? Aside from these comments, LGTM.

File: src/interpolation.jl [hunk 1]
```diff
@@ -147,6 +150,18 @@ function reverse(tinterp::TaylorInterpolant{T,U,N}) where {T<:Real, U<:Number, N
     return TaylorInterpolant(tinterp_rev_t0, tinterp_rev_t, tinterp_rev_x)
 end
 
+@doc raw"""
+    selecteph(eph::TaylorInterpolant{T, U, N}, i::Int) where {T <: Real, U <: Number, N}
+
+Return a `TaylorInterpolant` with only the ephemeris of the `i`-th body. 
+"""
+function selecteph(eph::TaylorInterpolant{T, U, N}, i::Int) where {T <: Real, U <: Number, N}
+    N = numberofbodies(eph)
```
Comment (lbenet) [inline src/interpolation.jl:159]: Is the `N` in the type-parameters of `eph` the same as the `N` obtained by `numberofbodies(eph)`? If so, I suggest to use the value from the type-parameter; if not, then, use `numberofbodies(eph)`, but perhaps use a different variable name.

Comment (PerezHz) [reply src/interpolation.jl:159]: IIUC, `N` is only used in the call to `nbodyind` to retrieve the indices. If so, then actually the `N` as a type parameter would not be needed, right?

**Comment by PerezHz** [comment]
Thanks @LuEdRaMo for this contribution, this is indeed really helpful! I agree with @lbenet that adding a test would be really nice, otherwise aside from the minor thing with the `N` as a type parameter, LGTM to me too!

File: src/interpolation.jl (hunk 1 above)
Comment (LuEdRaMo) [reply src/interpolation.jl]: Sorry for the typo, the function only works for `TaylorInterpolant{T, U, 2}` and `N` must be the result of `numberofbodies`.

**Comment by LuEdRaMo** [comment]
@PerezHz @lbenet a few comments:
- To include an automated test PE should have a way to accesss a ephemeris file. One option would be to bind an artifact just like in the most recent PR of NEOs. What do you think?
- Should we bump TaylorSeries and TaylorIntegration versions?

**Comment by PerezHz** [comment]
> To include an automated test PE should have a way to accesss a ephemeris file. One option would be to bind an artifact just like in the most recent PR of NEOs. What do you think?

That's a good point! For the tests I'd say we could try generating on the fly a very short ephemeris, say, with two or three time-steps, maybe even without any asteroids, and then we could readily test this function. If this is not too much work to add, and the tests do not become terribly slow, then I'd say we also have a ways to test the actual integrator, which can be really helpful to have.

> Should we bump TaylorSeries and TaylorIntegration versions?

If things don't break here, then I'm in favor!

**Comment by lbenet** [comment]
I agree with @PerezHz: I think a test-integration should cover two weeks or at most a month (fwd and bkw), in the sense that it should be fast to be obtained. I'm also in favor of upgrading TS and TI; there shouldn't be any problem.

What about Julia v1.9?

**Comment by PerezHz** [comment]
I would say it's definitely worth it to migrate from SnoopPrecompile to PrecompileTools and check whether we see some improvements in TTFX on julia1.9 but probably we can tackle that in a separate PR?

**Comment by LuEdRaMo** [comment]
1.- I did some test integrations and bumping TS and TI versions do not affect the results, although we must remember that they will be affected by https://github.com/PerezHz/PlanetaryEphemeris.jl/issues/10 between Julia versions.
2.- Currently, the main limitation to adding a test that generates the ephemeris on the fly is that `scripts/integrate_ephemeris.jl` takes a long time to start. I've been playing a bit with `PrecompileTools.jl` and with a simple block of code the package precompilation time is ∼40 min but after that TTFX is reduced significantly. 
3.- If you agree, I can bump PE minor version and let the issue with `PrecompileTools.jl` for another PR. If that is the case, this might be ready to merge...

**Comment by lbenet** [comment]
> 1.- I did some test integrations and bumping TS and TI versions do not affect the results, although we must remember that they will be affected by #10 between Julia versions.

My guess is that you used Julia v1.8 and compared it with the results obtained by the *old* versions of TS and TI using still Julia v1.8, right? 

What about differences by using Julia v1.9?

> 2.- Currently, the main limitation to adding a test that generates the ephemeris on the fly is that `scripts/integrate_ephemeris.jl` takes a long time to start. I've been playing a bit with `PrecompileTools.jl` and with a simple block of code the package precompilation time is ∼40 min but after that TTFX is reduced significantly.

I agree. I think we should begin with `PrecompileTools.jl` for TS, and perhaps latter for TI, in specific cases. That might help reducing the precompilation times further. I've opened #13 as a remainder...

> 3.- If you agree, I can bump PE minor version and let the issue with `PrecompileTools.jl` for another PR. If that is the case, this might be ready to merge...

I agree; let's us wait for the opinion of @PerezHz

**Comment by PerezHz** [comment]
I agree this PR is ready to be merged practically as it is now (modulo bumping the minor version, which I agree with), and then we can tackle both adding `PrecompileTools.jl` and some unit tests in another PR.

> If you agree, I can bump PE minor version and let the issue with PrecompileTools.jl for another PR. If that is the case, this might be ready to merge...

Feel free to bump the minor version, and merge! 😄

**Comment by LuEdRaMo** [comment]
> My guess is that you used Julia v1.8 and compared it with the results obtained by the _old_ versions of TS and TI using still Julia v1.8, right?
> 
> What about differences by using Julia v1.9?
> 
Using *older/newer* versions of TS and TI makes no difference on the results. Meanwhile, using Julia *pre/post* v1.7 is affected by https://github.com/PerezHz/PlanetaryEphemeris.jl/issues/10

**Pull request merged**